docs/concepts/node-agent.md

```diff
@@ -0,0 +1,59 @@
+# Gardener Node Agent
+
+The goal of the `gardener-node-agent` is to bootstrap a machine into a worker node and maintain node-specific components, which run on the node and are unmanaged by Kubernetes (e.g. the controller runtime, the kubelet service, ...).
+
+It effectively is a Kubernetes controller deployed onto the worker node.
+
+## Basic Design
+
+In this section it is described how the `gardener-node-agent` works, what its responsibilities are and how it is installed onto the worker node.
+
+To install the `gardener-node-agent` onto a worker node, there is a very small bash script called `gardener-node-init.sh`, which is installed on the node with cloud-init data. This script's sole purpose is downloading and starting the `gardener-node-agent`. The binary artifact is downloaded as an [OCI artifact](https://github.com/opencontainers/image-spec/blob/main/manifest.md), removing the `docker` dependency on a worker node. At the beginning, two architectures of the `gardener-node-agent` are supported: `amd64` and `x86`. In the same manner, the kubelet has to be provided as an OCI artifact.
```

Comment:
@majst01 can you have a look on this?

Reply:
@ScheererJ we discussed this together with the folks responsible for your CI build pipeline whether we can move to OCI artefacts instead of docker-images. I would prefer to do so, but this will need that we also migrate existing hyperkube images to OCI images as well which means double the storage.

With OCI images multi arch support is still possible.

Reply:
@majst01 I do not understand your point.

From my point of view, there are two supported architectures in the gardener universe. Those are `amd64` (sometimes also referred to as `x86-64`) and `arm64`. `x86` refers to 32-bit Intel/AMD processor architecture, which is less commonly used nowadays. This is why I wonder if you want to introduce an old platform here or simply used the wrong name.

Reply:
I assume that was a typo. As we haven't implemented the build process of the node agent, I'll remove the mention of supported platforms. The node agent should always support the same architectures as Gardener itself